parquet-common/src/main/java/org/apache/parquet/VersionParser.java

```diff
@@ -31,7 +31,7 @@
  */
 public class VersionParser {
   // example: parquet-mr version 1.8.0rc2-SNAPSHOT (build ddb469afac70404ea63b72ed2f07a911a8592ff7)
-  public static final String FORMAT = "(.+) version ((.*) )?\\(build ?(.*)\\)";
+  public static final String FORMAT = "(.+) version( |$)?(.*?)( |$)?(\\(build ?(.*)\\))?";
```

Comment:
Truth be told, I think using recursive descent would probably a much more maintainable solution to this, but I wanted to make the least amount of changes. What do you think?